application/app/controllers/dataverse/datasets_controller.rb

```diff
@@ -107,4 +107,27 @@ def search_files_page
       return
     end
   end
+
+  def redirect_back_to_app(fallback: root_path, **options)
+    if internal_referer?
+      redirect_back(fallback_location: fallback, **options)
+    else
+      redirect_to(fallback, **options)
+    end
+  end
+
+  def internal_referer?
```

Comment:
To make the name aligned to the behaviour, I would rename it to `redirect_back?`